### ISSUE TITLE
Cumstom quantized-mesh terrain not rendered on higher levels

### ISSUE THREAD

**Issue by JackD83** [AnalyticalGraphicsInc/cesium#7963]
We upgraded from Cesium 1.56.1 to 1.58 and our terrain is disappearing on higher zoom levels.
We use a custom terrain server with quantized-mesh-1.0 to generate world wide terrain data up to 10cm

In some rare cases, I do get terrain data, and from there on its working fine. Deleting the browser cache and the problem re-appears.

Sandcastle example: 
Our terrain server is not public, but the test was very basic:

var viewer = new Cesium.Viewer('cesiumContainer');
viewer.terrainProvider = new Cesium.CesiumTerrainProvider({url:"http://localhost:8281/terrain/"});     

The generated layer.json can be found here:
[https://pastebin.com/d4m9KHF5](url)

To see the sandcastle example, I made a small video

[https://youtu.be/F4fsKBxFY-4](url)

Browser:
Chrome 75.0.3770.100

**Comment by OmarShehata** [comment]
Thanks for reporting this @JackD83 . Can you test if this happens in:

* 1.57 - this is the version where a large terrain refactoring was introduced to significantly speed up performance, it may be a cause of the issue here
* The current master branch - there were some recent fixes with image requests, and I want to confirm that this hasn't been fixed yet.

Does the network tab show any errors when everything disappears like that? Can you try setting the [Globe.fillHighlightColor](https://cesiumjs.org/Cesium/Build/Documentation/Globe.html?classFilter=GLobe#fillHighlightColor) to something like red (this option is only available in 1.57+). Fill tiles are placeholder tiles created while waiting for terrain requests, so if you see them highlighted for a long period of time, then that's a bug.

**Comment by JackD83** [comment]
I did some new tests.

- 1.57 is also affected
- current master has the same problem (commit f41cc3f8816aaeba8c081bdc4e60c43a9356ce21)

I set Globe.fillHighlightColor to red and I see the red tiles on lower levels popping up. There are no red tiles on higher levels and the terrain disappears. There are no tiles requested.

I noticed that I can get closer to the terrain in parts of the world where I do not have higher level terrain

**Comment by gunmiosb** [comment]
I have the same issue with Cesium 1.60 Release

I don't think the parsing of the [layer.json](https://github.com/AnalyticalGraphicsInc/cesium/files/3565499/layer.json.txt) is the reason, because terrainProvider.getTileDataAvailable returns true for the missing tiles.

I discovered that tiles on the west side of the level 0 tiles are not loaded from level 9 on. And the farther i go to the east of the level 0 tile the more i can zoom in.

I made a video of the behavior:
[https://youtu.be/eJg3E-PTn1g](https://youtu.be/eJg3E-PTn1g)

Cesium doesn't request the missing tiles.

My Sandcastle Example:
[Sandcastle](https://cesiumjs.org/Cesium/Build/Apps/Sandcastle/#c=ZVDRSgMxEPyVkJdeQRJ8rdciVEFBUVR8ykua2/YWt8mR5O5axX8317Ta1kDIzu7M7JBOe9Yh9ODZlFno2RwCtmvxvusVI7ODc2ejRgt+NL5StkdbuV7k0a84F2mcCxHB+6R59q7D6tw+P2+nlOKr9TRRvI6xmUhJzmiqXYhybyUV/x7/LYBNBFsVe8fczODehgZMdP4RN2iPJMGABbEitwCxRKI7XNWUbpw7ckPEQ7wBipfbmyTlF7wMcUswU5alc43rxvnIUtZCiJRt3ZCOEOSiNR8QhQlh2FjKg6issGNYTRU/+0zFmSEdQposW6JX/ATFZ6VM/BMZOV2hXT114ElvB0p9OXvITSFEKRP8r4rO0UL7I8cf)

**Comment by OmarShehata** [comment]
Thanks @gunmiosb . How was this terrain generated? Is there any chance you're able to share the tileset that reproduces this? You can email it to omar@cesium.com if you can't share it publicly.

**Comment by OmarShehata** [comment]
Thanks for emailing me a sample to try @gunmiosb . I can confirm this issue happens in 1.57, but not in 1.56, which shows that it was likely the big terrain and imagery overhaul in this PR https://github.com/AnalyticalGraphicsInc/cesium/pull/7061 that caused it. 

The next step here I think would be to try to `git bisect` between 1.56 and 1.57 to determine what commit introduced the bug, which will help shed some light on either what needs to change in CesiumJS or the loaded terrain. If either if you are able to try that, it would be greatly appreciated.

@JackD83 can you confirm that the issue is not present in 1.56?

**Comment by JackD83** [comment]
@OmarShehata Yes, 1.56.1 works fine and its the version we are currently stuck at

**Comment by JackD83** [comment]
I finally had the time to do the bisect.  [This](https://github.com/AnalyticalGraphicsInc/cesium/commit/b96c5a7a0978888fe14e2d2c1bf16ab1cb4debfd#diff-b82d964ca6c373e9d1246973ec43c947R906) commit is breaking our trerrain

Removing `surfaceTile.occludeePointInScaledSpace = Cartesian3.clone(mesh.occludeePointInScaledSpace, surfaceTile.occludeePointInScaledSpace);` from GlobeSurfaceTile.js

Disabling the check `if (defined(rectangle) && rectangle.width < CesiumMath.PI_OVER_TWO + CesiumMath.EPSILON5) {` from  GlobeSurfaceTileProvider.js

makes the terrain work again. Question is what our terrain does wrong at that point
